src/Umbraco.Web.Common/AspNetCore/AspNetCoreHostingEnvironment.cs

```diff
@@ -26,6 +23,7 @@ public AspNetCoreHostingEnvironment(IHostingSettings hostingSettings, IWebHostEn
             ApplicationPhysicalPath = webHostEnvironment.ContentRootPath;
 
             ApplicationVirtualPath = "/"; //TODO how to find this, This is a server thing, not application thing.
+            ApplicationServerAddress = "https://localhost:44354"; // TODO how to find this?
```

Comment:
It does not need to be part of IHostingEnvironment interface, but IMO it is an okay fit based on the name of the interface?

Do you have ideas to how to get this value, as far as I can see, the only safe way, is as a config value, because reverse proxies will never be handled otherwise.

Reply:
We already offer this as part of config: `IWebRoutingSettings.UmbracoApplicationUrl`. Perhaps this config could/should move elsewhere. But i guess the real question is why we need this value before a request begins? Currently in v8 we don't. We dynamically set this value based on the initial request if the `IWebRoutingSettings.UmbracoApplicationUrl` isn't specified. (there used to be a 3rd place where this was configured https://github.com/umbraco/Umbraco-CMS/blob/v8/contrib/src/Umbraco.Core/Sync/IServerRegistrar.cs#L27 ... but we've thankfully removed that since this should just be consolidated to one place)

Reply:
In v8, we called `EnsureApplicationUrl` on each request, that set the application url, if it was changed. 

This was pretty nasty, as we casted the IRuntimeState to RuntimeState just to call that method 😭 

Now I implemented a `GetApplicationUrl()` on `IRequestAccessor` that does this whenever the application url is used. In general this is `< 1` time per request.

Reply:
The original code in `EnsureApplicationUrl` had a fixme comment. I currently keep this comment, but i'm unsure if this is still relevant

Reply:
Yep, the problem is when we dynamically assign this URL which has issues with things like slot swapping and a bunch of other things. I think the reality is that for people hosting on Azure or doing things where the URL can actually change hosts, etc... they will need to configure the URL. I'll have a look at the changes.